File: src/main/scala/com/kapilkoju/autopos/party/repo/VehicleRepo.scala
package com.kapilkoju.autopos.party.repo

import com.kapilkoju.autopos.kernel.service.AuditableBaseRepository
import com.kapilkoju.autopos.party.domain.Vehicle
import org.springframework.data.domain.{Page, Pageable}

trait VehicleRepo extends AuditableBaseRepository[Vehicle] {

  def findByNumberIgnoreCaseContainingOrOwnerNameIgnoreCaseContaining(numberQuery: String,
                                                                      ownerNameQuery: String,
                                                                      pageable: Pageable
                                                                     ): Page[Vehicle]

}

File: src/main/kotlin/kotlin/com/kapilkoju/autopos/party/service/VehicleRepo.kt
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.kernel.service.AuditableRepository
import com.kapilkoju.autopos.party.domain.Vehicle
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable


interface VehicleRepo : AuditableRepository<Vehicle> {

    fun findByNumberIgnoreCaseContainingOrOwnerNameIgnoreCaseContaining(numberQuery: String,
                                                                        ownerNameQuery: String,
                                                                        pageable: Pageable
    ): Page<Vehicle>

}
